snafu/benchmarks/coremarkpro/coremarkpro.py

```diff
@@ -0,0 +1,207 @@
+                    if re.search(r'^\d+', line):
+                        cols = re.split(r'\s+', line.rstrip())
+                        converted_cols = [func(val) for func, val in zip(types, cols)]
+                        record = dict(zip(headers, converted_cols))
+                        record['type'] = run_type
+                        record['starttime'] = self.convert_coremark_timestamp(run_starttime)
+                        results.append(record)
+
+            return results
+
+    def parse_marks(self):
+        headers = ['name',  'multicore', 'singlecore', 'scaling']
+        types = [str, float, float, float]
+        test_results = []
+        with open(self.config.path + 'builds/linux64/gcc64/logs/linux64.gcc64.mark', 'rt') as file:
+            table_name = ""
+            results = {}
+            for line in file:
+                line = line.rstrip()
+                if not line:
+                    continue
+
+                # Find where the table starts and skips the fluff
+                if "RESULTS TABLE" in line:
+                    table_name = line.split(' ')[0].lower()
+                    while True:
+                        # Exit out of loop once it finds the table delimiter
+                        if('---' in next(file)):
+                            break
+                    continue
+
+                cols = re.split(r'\s+', line.rstrip())
+                converted_cols = [func(val) for func, val in zip(types, cols)]
+                record = dict(zip(headers, converted_cols))
+                record['type'] = table_name
+                test_results.append(record)
+
+        return test_results
+
+    def submit_results(self, sample_starttime) -> BenchmarkResult:
+
+        test_config = {
+                "workload": self.config.workload,
+                "context": self.config.context,
+                }
+        result_summary = {
+                #"clustername": self.config.cluster_name,
+                "date": sample_starttime,
+                "sample": self.config.sample,
+                "test_config": test_config,
+                }
+
+        test_results = {
+                "summary_results": self.parse_marks(),
+                "raw_results": self.parse_raw_logs()
+                }
+
+        return self.create_new_result(
+                data=test_results,
+                config=result_summary,
+                tag="results",
+                )
+
+    def convert_coremark_timestamp(self, timestamp):
+        from dateutil import tz
+        time_obj = datetime.strptime(timestamp, '%y%j:%H:%M:%S')
+        utc_tz = tz.gettz('UTC')
+
+        return (time_obj.astimezone(utc_tz)).strftime("%Y-%m-%dT%H:%M:%S.%fZ")   
+
+
+    """ Override member functions from Benchmark class"""
+
+    def setup(self) -> bool:
+
+        # Parse the command line args
+        self.config.parse_args()
+
+        # Sets up defaults for the required variables
```

Comment:
Following are required and sets the default values, I copied the defaults from the sysbench wrapper